R/pkg/inst/worker/worker.R

```diff
@@ -84,6 +84,10 @@ broadcastElap <- elapsedSecs()
 # as number of partitions to create.
 numPartitions <- SparkR:::readInt(inputCon)
 
+# If true, working for RDD
+# If false, working for DataFrame
+isDataFrame <- as.logical(SparkR:::readInt(inputCon))
```

Comment:
generally, I agree. it is possible though an user could have an initialization or profile file that inadvertently loads a mismatch version of SparkR..